Reddit/script.js

```diff
@@ -0,0 +1,42 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const fetchButton = document.getElementById("fetchButton");
+  const subredditNameInput = document.getElementById("subredditName");
+  const postContainer = document.getElementById("postContainer");
+
+  fetchButton.addEventListener("click", function () {
+    const subredditName = subredditNameInput.value;
+
+    fetch(`https://www.reddit.com/r/${subredditName}/.json`)
+      .then((response) => response.json())
+      .then((data) => {
+        postContainer.innerHTML = "";
+
+        const posts = data.data.children.slice(0, 10);
+
+        posts.forEach((post) => {
+          const postElement = document.createElement("div");
+          postElement.classList.add("post");
+
+          const titleElement = document.createElement("h2");
+          titleElement.textContent = post.data.title;
+
+          const imageElement = document.createElement("img");
+          imageElement.src = post.data.thumbnail;
+
+          const linkElement = document.createElement("a");
+          linkElement.href = `https://www.reddit.com${post.data.permalink}`;
```

Comment:
You wouldn't need the leading string if you pointed to post.data.url. The url includes "https://www.reddit.com"